solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2ClusterStateProvider.java

```diff
@@ -19,33 +19,34 @@
 
 import java.io.IOException;
 import java.util.List;
-import org.apache.solr.client.solrj.SolrClient;
 
-public class Http2ClusterStateProvider extends BaseHttpClusterStateProvider {
-  final Http2SolrClient httpClient;
-  final boolean closeClient;
+public class Http2ClusterStateProvider<C extends HttpSolrClientBase>
+    extends BaseHttpClusterStateProvider {
+  final C httpClient;
 
-  public Http2ClusterStateProvider(List<String> solrUrls, Http2SolrClient httpClient)
-      throws Exception {
-    this.httpClient = httpClient == null ? new Http2SolrClient.Builder().build() : httpClient;
-    this.closeClient = httpClient == null;
+  public Http2ClusterStateProvider(List<String> solrUrls, C httpClient) throws Exception {
+    if (httpClient == null) {
```

Comment:
Up to you but consider the conciseness of using the same line that sets httpClient to use `Object.requireNonNull(httpClient,"httpClient is required")`.  True that it's an NPE and not IAE but...